lib/service_provider.rb

```diff
@@ -36,14 +36,27 @@ def cert
     @cert ||= File.read("#{cert_dir}#{host_attributes['cert']}.crt")
   end
 
+  def encrypt_responses?
+    block_encryption != 'none'
+  end
+
   def block_encryption
-    host_attributes['block_encryption']
+    host_attributes['block_encryption'] || 'none'
   end
 
   def key_transport
     'rsa-oaep-mgf1p'
   end
 
+  def encryption_opts
+    return nil unless encrypt_responses?
```

Comment:
Could this be simplified to `return nil unless block_encryption`, then removing `|| 'none'` from `block_encrpyption` and removing the `encrypt_responses?` method?